File: solutions/dart/anagram/1/lib/anagram.dart
class Anagram {
  String _sortString(String input){
    return (input.split('')..sort()).join('');
  }  // Put your code here

  List<String> findAnagrams(String word, List<String> candidates) {
    final sortedWord = _sortString(word.toLowerCase());

    var anagrams=candidates.where((candidate){
      if(candidate.toLowerCase()==word.toLowerCase()){
        return false;
      }
      return _sortString(candidate.toLowerCase()) == sortedWord.toLowerCase();
    }).toList();
    
    return anagrams;
  }
}

File: solutions/python/anagram/1/anagram.py
def find_anagrams(word, candidates):
    res=[]

    for candidate in candidates:
        if word.lower()==candidate.lower(): # The same word
            continue
        else:
            char_list_word=list(sorted(word.lower()))
            cand_list_word=list(sorted(candidate.lower()))

            if char_list_word==cand_list_word:
                res.append(candidate)
    
    return res


